### ISSUE TITLE
[TEST-FAILURE] client.ClientRegressionWithMockNetworkTest timeout failures

### ISSUE THREAD

**Issue by hasancelik** [hazelcast/hazelcast#6153]
**ClientRegressionWithMockNetworkTest.testOperationRedo**
**ClientRegressionWithMockNetworkTest.testGithubIssue3557**

```
org.junit.runners.model.TestTimedOutException: test timed out after 60000 milliseconds
	at sun.misc.URLClassPath$JarLoader.validIndex(URLClassPath.java:1032)
	at sun.misc.URLClassPath$JarLoader.getResource(URLClassPath.java:1166)
	at sun.misc.URLClassPath$JarLoader.getResource(URLClassPath.java:1077)
	at sun.misc.URLClassPath$JarLoader.findResource(URLClassPath.java:1047)
	at sun.misc.URLClassPath.findResource(URLClassPath.java:298)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:901)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:899)
	at java.security.AccessController.doPrivileged(AccessController.java:252)
```

**ClientRegressionWithMockNetworkTest.com.hazelcast.client.ClientRegressionWithMockNetworkTest**

```
java.lang.IllegalStateException: Instances haven't been shut down: [HazelcastInstance{name='_hzInstance_14_dev', node=Address[127.0.0.1]:5022}, HazelcastInstance{name='_hzInstance_27_dev', node=Address[127.0.0.1]:5027}]
	at com.hazelcast.test.AbstractHazelcastClassRunner$1.evaluate(AbstractHazelcastClassRunner.java:189)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-IbmJDK1.7/692/

**Comment by sancar** [comment]
I did not do a detailed research but it looks like highly parallel tests are somehow causing deadlock in classLoader related parts.
@serkan-ozal Do you have an idea ?

**Comment by hasancelik** [comment]
**I updated ibm version for this build but still we are getting this failure.**

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-IbmJDK1.7/706/

**Comment by hasancelik** [comment]
**ClientRegressionWithMockNetworkTest.com.hazelcast.client.ClientRegressionWithMockNetworkTest**

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-OpenJDK8/com.hazelcast$hazelcast-client-new/695/testReport/junit/com.hazelcast.client/ClientRegressionWithMockNetworkTest/com_hazelcast_client_ClientRegressionWithMockNetworkTest/

**Comment by hasancelik** [comment]
**ClientRegressionWithMockNetworkTest.com.hazelcast.client.ClientRegressionWithMockNetworkTest**

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x/com.hazelcast$hazelcast-client-new/3765/testReport/junit/com.hazelcast.client/ClientRegressionWithMockNetworkTest/com_hazelcast_client_ClientRegressionWithMockNetworkTest/

**Comment by hasancelik** [comment]
**testOperationRedo**
```
org.junit.runners.model.TestTimedOutException: test timed out after 60000 milliseconds
	at java.util.jar.JarFile$JarEntryIterator.nextElement(JarFile.java:342)
	at java.util.jar.JarFile$JarEntryIterator.nextElement(JarFile.java:323)
	at sun.misc.URLClassPath$JarLoader.validIndex(URLClassPath.java:1072)
	at sun.misc.URLClassPath$JarLoader.getResource(URLClassPath.java:1205)
	at sun.misc.URLClassPath$JarLoader.getResource(URLClassPath.java:1117)
	at sun.misc.URLClassPath$JarLoader.findResource(URLClassPath.java:1087)
	at sun.misc.URLClassPath.findResource(URLClassPath.java:309)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:917)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:915)
	at java.security.AccessController.doPrivileged(AccessController.java:620)
	at java.net.URLClassLoader.findResource(URLClassLoader.java:914)
	at java.lang.ClassLoader.getResource(ClassLoader.java:541)
	at java.net.URLClassLoader.getResourceAsStream(URLClassLoader.java:362)
	at org.apache.xml.dtm.SecuritySupport$6.run(Unknown Source)
	at java.security.AccessController.doPrivileged(AccessController.java:594)
	at org.apache.xml.dtm.SecuritySupport.getResourceAsStream(Unknown Source)
	at org.apache.xml.dtm.ObjectFactory.findJarServiceProviderName(Unknown Source)
	at org.apache.xml.dtm.ObjectFactory.lookUpFactoryClassName(Unknown Source)
	at org.apache.xml.dtm.ObjectFactory.lookUpFactoryClass(Unknown Source)
	at org.apache.xml.dtm.ObjectFactory.createObject(Unknown Source)
	at org.apache.xml.dtm.ObjectFactory.createObject(Unknown Source)
	at org.apache.xml.dtm.DTMManager.newInstance(Unknown Source)
	at org.apache.xpath.XPathContext.<init>(Unknown Source)
	at org.apache.xpath.jaxp.XPathExpressionImpl.evaluate(Unknown Source)
	at org.apache.xpath.jaxp.XPathImpl.evaluate(Unknown Source)
	at com.hazelcast.config.AbstractConfigBuilder.traverseChildsAndReplaceVariables(AbstractConfigBuilder.java:135)
	at com.hazelcast.config.AbstractConfigBuilder.traverseChildsAndReplaceVariables(AbstractConfigBuilder.java:154)
	at com.hazelcast.config.AbstractConfigBuilder.traverseChildsAndReplaceVariables(AbstractConfigBuilder.java:154)
	at com.hazelcast.config.AbstractConfigBuilder.traverseChildsAndReplaceVariables(AbstractConfigBuilder.java:154)
	at com.hazelcast.config.AbstractConfigBuilder.process(AbstractConfigBuilder.java:73)
	at com.hazelcast.config.XmlConfigBuilder.parseAndBuildConfig(XmlConfigBuilder.java:218)
	at com.hazelcast.config.XmlConfigBuilder.build(XmlConfigBuilder.java:202)
	at com.hazelcast.config.XmlConfigBuilder.build(XmlConfigBuilder.java:195)
	at com.hazelcast.instance.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:129)
	at com.hazelcast.test.TestHazelcastInstanceFactory.newHazelcastInstance(TestHazelcastInstanceFactory.java:110)
	at com.hazelcast.test.TestHazelcastInstanceFactory.newHazelcastInstance(TestHazelcastInstanceFactory.java:97)
	at com.hazelcast.client.ClientRegressionWithMockNetworkTest.testOperationRedo(ClientRegressionWithMockNetworkTest.java:160)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-IbmJDK1.8/com.hazelcast$hazelcast-client-legacy/67/

**Comment by mesutcelik** [comment]
@hasancelik ,
I think we should create a new issue instead of reopening current issue. The issue in the comment history is related to new client protocol but latest failure is related to legacy client....

**Comment by cangencer** [comment]
looks like some kind of deadlock when the JDK is reading from JARs

```
"Time-limited test" 
	java.lang.Thread.State: BLOCKED on lock=java.util.jar.JarFile@8b7a77f6 owned by Time-limited test id=12781
		at java.util.zip.ZipFile$ZipEntryIterator.next(ZipFile.java:557)
		at java.util.zip.ZipFile$ZipEntryIterator.nextElement(ZipFile.java:553)
		at java.util.zip.ZipFile$ZipEntryIterator.nextElement(ZipFile.java:534)
		at java.util.jar.JarFile$JarEntryIterator.next(JarFile.java:333)
		at java.util.jar.JarFile$JarEntryIterator.nextElement(JarFile.java:342)
		at java.util.jar.JarFile$JarEntryIterator.nextElement(JarFile.java:323)
		at sun.misc.URLClassPath$JarLoader.validIndex(URLClassPath.java:1072)
		at sun.misc.URLClassPath$JarLoader.getResource(URLClassPath.java:1205)
		at sun.misc.URLClassPath$JarLoader.getResource(URLClassPath.java:1117)
		at sun.misc.URLClassPath$JarLoader.findResource(URLClassPath.java:1087)
		at sun.misc.URLClassPath.findResource(URLClassPath.java:309)
		at java.net.URLClassLoader$1.run(URLClassLoader.java:917)
		at java.net.URLClassLoader$1.run(URLClassLoader.java:915)
		at java.security.AccessController.doPrivileged(AccessController.java:620)
		at java.net.URLClassLoader.findResource(URLClassLoader.java:914)
		at java.lang.ClassLoader.getResource(ClassLoader.java:541)
		at java.net.URLClassLoader.getResourceAsStream(URLClassLoader.java:362)
		at org.apache.xml.dtm.SecuritySupport$6.run(Unknown Source)
		at java.security.AccessController.doPrivileged(AccessController.java:594)
		at org.apache.xml.dtm.SecuritySupport.getResourceAsStream(Unknown Source)
		at org.apache.xml.dtm.ObjectFactory.findJarServiceProviderName(Unknown Source)
		at org.apache.xml.dtm.ObjectFactory.lookUpFactoryClassName(Unknown Source)
		at org.apache.xml.dtm.ObjectFactory.lookUpFactoryClass(Unknown Source)
		at org.apache.xml.dtm.ObjectFactory.createObject(Unknown Source)
		at org.apache.xml.dtm.ObjectFactory.createObject(Unknown Source)
		at org.apache.xml.dtm.DTMManager.newInstance(Unknown Source)
		at org.apache.xpath.XPathContext.<init>(Unknown Source)
		at org.apache.xpath.jaxp.XPathExpressionImpl.evaluate(Unknown Source)
		at org.apache.xpath.jaxp.XPathImpl.evaluate(Unknown Source)
		at com.hazelcast.config.AbstractConfigBuilder.traverseChildsAndReplaceVariables(AbstractConfigBuilder.java:135)
		at com.hazelcast.config.AbstractConfigBuilder.traverseChildsAndReplaceVariables(AbstractConfigBuilder.java:154)
		at com.hazelcast.config.AbstractConfigBuilder.traverseChildsAndReplaceVariables(AbstractConfigBuilder.java:154)
		at com.hazelcast.config.AbstractConfigBuilder.process(AbstractConfigBuilder.java:73)
		at com.hazelcast.config.XmlConfigBuilder.parseAndBuildConfig(XmlConfigBuilder.java:218)
		at com.hazelcast.config.XmlConfigBuilder.build(XmlConfigBuilder.java:202)
		at com.hazelcast.config.XmlConfigBuilder.build(XmlConfigBuilder.java:195)
		at com.hazelcast.instance.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:129)
		at com.hazelcast.test.TestHazelcastInstanceFactory.newHazelcastInstance(TestHazelcastInstanceFactory.java:110)
		at com.hazelcast.test.TestHazelcastInstanceFactory.newHazelcastInstance(TestHazelcastInstanceFactory.java:97)
		at com.hazelcast.client.ClientRegressionWithMockNetworkTest.testMapDestroyIssue764(ClientRegressionWithMockNetworkTest.java:232)
		at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:95)
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
		at java.lang.reflect.Method.invoke(Method.java:507)
		at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
		at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
		at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
		at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
		at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
		at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
		at java.util.concurrent.FutureTask.run(FutureTask.java:277)
		at java.lang.Thread.run(Thread.java:785)
```
and
```
"Time-limited test" 
	java.lang.Thread.State: RUNNABLE
		at java.util.zip.ZipFile.getNextEntry(Native Method)
		at java.util.zip.ZipFile.access$500(ZipFile.java:76)
		at java.util.zip.ZipFile$ZipEntryIterator.next(ZipFile.java:562)
		at java.util.zip.ZipFile$ZipEntryIterator.nextElement(ZipFile.java:553)
		at java.util.zip.ZipFile$ZipEntryIterator.nextElement(ZipFile.java:534)
		at java.util.jar.JarFile$JarEntryIterator.next(JarFile.java:333)
		at java.util.jar.JarFile$JarEntryIterator.nextElement(JarFile.java:342)
		at java.util.jar.JarFile$JarEntryIterator.nextElement(JarFile.java:323)
		at sun.misc.URLClassPath$JarLoader.validIndex(URLClassPath.java:1072)
		at sun.misc.URLClassPath$JarLoader.getResource(URLClassPath.java:1205)
		at sun.misc.URLClassPath$JarLoader.getResource(URLClassPath.java:1117)
		at sun.misc.URLClassPath$JarLoader.findResource(URLClassPath.java:1087)
		at sun.misc.URLClassPath.findResource(URLClassPath.java:309)
		at java.net.URLClassLoader$1.run(URLClassLoader.java:917)
		at java.net.URLClassLoader$1.run(URLClassLoader.java:915)
		at java.security.AccessController.doPrivileged(AccessController.java:620)
		at java.net.URLClassLoader.findResource(URLClassLoader.java:914)
		at java.lang.ClassLoader.getResource(ClassLoader.java:541)
		at java.net.URLClassLoader.getResourceAsStream(URLClassLoader.java:362)
		at java.lang.ClassLoader.getSystemResourceAsStream(ClassLoader.java:788)
		at org.apache.xml.dtm.SecuritySupport$6.run(Unknown Source)
		at java.security.AccessController.doPrivileged(AccessController.java:594)
		at org.apache.xml.dtm.SecuritySupport.getResourceAsStream(Unknown Source)
		at org.apache.xml.dtm.ObjectFactory.findJarServiceProviderName(Unknown Source)
		at org.apache.xml.dtm.ObjectFactory.lookUpFactoryClassName(Unknown Source)
		at org.apache.xml.dtm.ObjectFactory.lookUpFactoryClass(Unknown Source)
		at org.apache.xml.dtm.ObjectFactory.createObject(Unknown Source)
		at org.apache.xml.dtm.ObjectFactory.createObject(Unknown Source)
		at org.apache.xml.dtm.DTMManager.newInstance(Unknown Source)
		at org.apache.xpath.XPathContext.<init>(Unknown Source)
		at org.apache.xpath.jaxp.XPathExpressionImpl.evaluate(Unknown Source)
		at org.apache.xpath.jaxp.XPathImpl.evaluate(Unknown Source)
		at com.hazelcast.config.AbstractConfigBuilder.traverseChildsAndReplaceVariables(AbstractConfigBuilder.java:135)
		at com.hazelcast.config.AbstractConfigBuilder.traverseChildsAndReplaceVariables(AbstractConfigBuilder.java:154)
		at com.hazelcast.config.AbstractConfigBuilder.traverseChildsAndReplaceVariables(AbstractConfigBuilder.java:154)
		at com.hazelcast.config.AbstractConfigBuilder.traverseChildsAndReplaceVariables(AbstractConfigBuilder.java:154)
		at com.hazelcast.config.AbstractConfigBuilder.process(AbstractConfigBuilder.java:73)
		at com.hazelcast.config.XmlConfigBuilder.parseAndBuildConfig(XmlConfigBuilder.java:218)
		at com.hazelcast.config.XmlConfigBuilder.build(XmlConfigBuilder.java:202)
		at com.hazelcast.config.XmlConfigBuilder.build(XmlConfigBuilder.java:195)
		at com.hazelcast.instance.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:129)
		at com.hazelcast.test.TestHazelcastInstanceFactory.newHazelcastInstance(TestHazelcastInstanceFactory.java:110)
		at com.hazelcast.test.TestHazelcastInstanceFactory.newHazelcastInstance(TestHazelcastInstanceFactory.java:97)
		at com.hazelcast.client.ClientRegressionWithMockNetworkTest.testGetDistributedObjectsIssue678(ClientRegressionWithMockNetworkTest.java:221)
		at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:95)
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
		at java.lang.reflect.Method.invoke(Method.java:507)
		at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
		at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
		at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
		at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
		at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
		at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
		at java.util.concurrent.FutureTask.run(FutureTask.java:277)
		at java.lang.Thread.run(Thread.java:785)
```

**Comment by cangencer** [comment]
We could put a synchronized block around XmlConfigBuilder.build() , what do you think @sancar ?
